gadgets/Makefile

```diff
@@ -21,5 +24,35 @@ all: $(GADGETS)
 		-I ../pkg/gadgets/internal/socketenricher/bpf/ \
 		-o $@
 
+# Networking gadgets
+#
+# We need <asm/types.h> and depending on Linux distributions, it is installed
+# at different paths:
+#
+# * Ubuntu, package linux-libc-dev:
+#   /usr/include/x86_64-linux-gnu/asm/types.h
+#
+# * Fedora, package kernel-headers
+#   /usr/include/asm/types.h
+#
+# Since Ubuntu does not install it in a standard path, add a compiler flag for
+# it.
+%_bpfel.bpf.o : %.bpf.c
+	CLANG_OS_FLAGS= ; \
+	if [ "$(shell grep -oP '^NAME="\K\w+(?=")' /etc/os-release)" == "Ubuntu" ]; then \
+		CLANG_OS_FLAGS="-I/usr/include/$(shell uname -m)-linux-gnu" ; \
+	fi ; \
+	docker run --rm \
+		--name ebpf-object-builder \
```

Comment:
It does not work for me when compiled without docker (on Fedora)... Hence using the docker builder.

Reply:
What error do you get?
Using the following worked fine for me on mint.

```Makefile
%_bpfel.bpf.o : %.bpf.c
	clang -target bpf -Wall -g -O2 -c $(filter %.c,$^) \
		-I ../pkg/gadgets/common/ \
		-I ../pkg/gadgets/internal/socketenricher/bpf/ \
		-o $@
```

I really think we should avoid using docker here now, it'll come later done in another layer.